samples/bluetooth/broadcast_audio_source/src/main.c

```diff
@@ -35,19 +35,80 @@ BUILD_ASSERT(CONFIG_BT_ISO_TX_BUF_COUNT >= TOTAL_BUF_NEEDED,
 #if defined(CONFIG_BAP_BROADCAST_16_2_1)
 
 static struct bt_bap_lc3_preset preset_active = BT_BAP_LC3_BROADCAST_PRESET_16_2_1(
-	BT_AUDIO_LOCATION_FRONT_LEFT, BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+	BT_AUDIO_LOCATION_FRONT_LEFT | BT_AUDIO_LOCATION_FRONT_RIGHT,
+	BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+
+#define BROADCAST_SAMPLE_RATE	16000
 
 #elif defined(CONFIG_BAP_BROADCAST_24_2_1)
 
 static struct bt_bap_lc3_preset preset_active = BT_BAP_LC3_BROADCAST_PRESET_24_2_1(
-	BT_AUDIO_LOCATION_FRONT_LEFT, BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+	BT_AUDIO_LOCATION_FRONT_LEFT | BT_AUDIO_LOCATION_FRONT_RIGHT,
+	BT_AUDIO_CONTEXT_TYPE_UNSPECIFIED);
+
+#define BROADCAST_SAMPLE_RATE	24000
 
 #endif
 
+#if defined(CONFIG_LIBLC3)
+#include "lc3.h"
+
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+#include <zephyr/usb/usb_device.h>
+#include <zephyr/usb/class/usb_audio.h>
+#include <zephyr/sys/ring_buffer.h>
+
+/* USB Audio Data is downsampled from 48kHz stereo to 16kHz mono when receiving data */
+#define USB_SAMPLE_RATE         48000
+#define USB_DOWNSAMPLE_RATE     BROADCAST_SAMPLE_RATE
+#define USB_FRAME_DURATION_US   1000
+#define USB_NUM_SAMPLES         ((USB_FRAME_DURATION_US * USB_DOWNSAMPLE_RATE) / USEC_PER_SEC)
+#define USB_BYTES_PER_SAMPLE    2
+#define USB_CHANNELS		2
```

Comment:
Are these values true for all USB connections?

Reply:
Currently yes, but that's because current USB Audio class is pretty much inflexible. Ideally it should be possible to just configure the `BROADCAST_SAMPLE_RATE` directly in USB Audio class and have the host to send the data at that rate (instead of hardcoded 48 kHz).

Reply:
I would suggest changing this once we have a flexible USB Audio device (and know the configs to look for)